lib/collection/src/shards/replica_set/snapshots.rs

```diff
@@ -82,6 +104,7 @@ impl ShardReplicaSet {
     pub async fn restore_local_replica_from(
```

Comment:
This function is humongous, spanning a few hundred lines. I think we should split it, but we can do so in a separate PR.

Reply:
+1